lib/Parse/Lexer.cpp

```diff
@@ -353,8 +353,13 @@ void Lexer::skipToEndOfLine(bool EatNewline) {
     case 0:
       // If this is a random nul character in the middle of a buffer, skip it as
       // whitespace.
-      if (CurPtr-1 != BufferEnd) {
+      switch (getNulCharacterKind(CurPtr - 1)) {
+      case NulCharacterKind::Embedded:
         diagnoseEmbeddedNul(Diags, CurPtr-1);
+        LLVM_FALLTHROUGH;
+      case NulCharacterKind::CodeCompletion:
+        continue;
+      case NulCharacterKind::BufferEnd:
         break;
```

Comment:
Could you move the task in the case of `NulCharacterKind::BufferEnd` into this branch? i.e.
```cpp
case NulCharacterKind::BufferEnd:
  // The last line of the file does not have a newline.
  --CurPtr;
  return;
}
```